internal/pkg/experimental/action/operator_uninstall.go

```diff
@@ -0,0 +1,70 @@
+package action
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	olmv1 "github.com/operator-framework/operator-controller/api/v1alpha1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/operator-framework/kubectl-operator/pkg/action"
+)
+
+type OperatorUninstall struct {
+	config *action.Configuration
+
+	Package string
+
+	Logf func(string, ...interface{})
+}
+
+func NewOperatorUninstall(cfg *action.Configuration) *OperatorUninstall {
+	return &OperatorUninstall{
+		config: cfg,
+		Logf:   func(string, ...interface{}) {},
+	}
+}
+
+func (u *OperatorUninstall) Run(ctx context.Context) error {
+	opKey := types.NamespacedName{Name: u.Package}
+	op := &olmv1.Operator{}
+	op.SetName(opKey.Name)
+	op.SetGroupVersionKind(olmv1.GroupVersion.WithKind("Operator"))
```

Comment:
Is this necessary when the object is already explicitly of type `olmv1.Operator`?